modules/150-user-authn/hooks/migration_groups_from_user_to_group_object_test.go

```diff
@@ -61,55 +88,94 @@ metadata:
 spec:
   email: user@example.com
   groups:
+  - admins
```

Comment:
I believe we should preserve group names which were valid metadata.names from the beginning. Thus we would not punish users who were lucky to name groups according to new concerns. They would not need to rename groups if they were named well.

E.g. 

- admins → admins
- Admins → admins-(somehash)
- -Admins → admins-(anotherhash)

Reply:
Good point, fixed it.